### ISSUE TITLE
$mdUtil.initOptionalProperties breaks with angular 1.4.1

### ISSUE THREAD

**Issue by mprokopowicz** [angular/material#3315]
Since this commit in angularjs: https://github.com/angular/angular.js/commit/a6339d30d1379689da5eec9647a953f64821f8b0
$mdUtil.initOptionalProperties could not relay on presence of attr in $attrs to determinate "hasKey" variable value.

As for now, this causes invalid defaults, for example in mdTabs.

I'm working on a PR with updated initOptionalProperties.

Also I think it will be much safer (and more readable) to explicit provide defaults in all components.

**Comment by ThomasBurleson** [comment]
> BTW, @mprokopowicz what process did you use to track the issue source back to `initOptionalProperties()`?

**Comment by mprokopowicz** [comment]
@ThomasBurleson: mdTabs in my app was broken after updating angular to 1.4.1.

TL;DR: chrome inspector and console.log (Or "dupa debugging" as we call this in Poland.) and investigation of latest commits to angular.js.

Long story:
We have quickly noticed that ```md-dynamic-height``` class was present on our tabs, by we don't use this one.

So I found this watcher in angular-material.js:
```
    $scope.$watch('dynamicHeight', function (value) {
      if (value) $element.addClass('md-dynamic-height');
      else $element.removeClass('md-dynamic-height');
    });
```
and simply found out that value always evaluate to ```true```.

Then I've checked latest commits to angular.js (I thought that angular 1.4.1 introduced some bug) and found commit which I've mention above.

Then it was obvious that it must be something with initOptionalProperties.
